dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -999,6 +1000,37 @@ def dpnp_inv(a):
     return b_f
 
 
+def dpnp_matrix_rank(A, tol=None, hermitian=False):
+    """
+    dpnp_matrix_rank(A, tol=None, hermitian=False)
+
+    Return matrix rank of array using SVD method.
+
+    """
+
+    if A.ndim < 2:
+        return (A != 0).any().astype(int)
+
+    S = dpnp_svd(A, compute_uv=False, hermitian=hermitian)
+
+    if tol is None:
+        tol = (
+            S.max(axis=-1, keepdims=True)
+            * max(A.shape[-2:])
+            * dpnp.finfo(S.dtype).eps
+        )
+    else:
+        if dpnp.is_supported_array_type(tol):
+            # Check that `a` and `tol` are allocated on the same device
+            # and have the same queue. Otherwise, `ValueError`` will be raised.
+            get_usm_allocations([A, tol])
+        else:
+            # Allocate dpnp.ndarray if tol is a scalar
+            tol = dpnp.array(tol, usm_type=A.usm_type, sycl_queue=A.sycl_queue)
+        tol = tol[..., None]
```

Comment:
What is the purpose of that?
I guess without that line we can remove block with lines 1023..1030, because
- `S > tol` works with a scalar
- `S > tol` returns an error if `S` and `tol` were allocated on different devices